eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/UnifiedSet.java

```diff
@@ -1282,8 +1282,20 @@ public boolean equals(Object object)
             return false;
         }
 
-        Set<?> other = (Set<?>) object;
-        return this.size() == other.size() && this.containsAll(other);
+        Set<?> s = (Set<?>) object;
```

Comment:
Hi @victornoel, this confused me at first but after digging around, I think I understand why you added the try catch. It's possible for two instances of `UnifiedSet` of different types to be compared, and it's possible `contains` can throw a `ClassCastException` in a `ChainedBucket`. I think this is possible a bug in `contains` and if it can be replicated in a test, should be fixed there. @mohrezaei @motlin @nikhilnanivadekar @prathasirisha What do you think?

Reply:
Actually, I added those try/catch because the javadoc from `Collection.containsAll` says they can optionally be thrown :)

Reply:
Well, would you look at that. I guess since it's part of the contract, there is nothing for us to fix in `contains` or `containsAll`.

Reply:
I don't think catching the exceptions is actually sensible. `this.containsAll` is not some random method on an unknown set implementation. The implementation is right here. It resolves to:
```
        if (iterable instanceof MutableCollection)
        {
            return ((MutableCollection<T>) iterable).allSatisfyWith(predicate, parameter);
        }
        if (iterable instanceof ArrayList)
        {
            return ArrayListIterate.allSatisfyWith((ArrayList<T>) iterable, predicate, parameter);
        }
        if (iterable instanceof RandomAccess)
        {
            return RandomAccessListIterate.allSatisfyWith((List<T>) iterable, predicate, parameter);
        }
        if (iterable != null)
        {
            return IterableIterate.allSatisfyWith(iterable, predicate, parameter);
        }
```
notice how it never actually calls `containsAll` (on the unknown collection). So unless those exceptions are possible from `allSatisfyWith`, I'd suggest reverting this bit.

Reply:
Also, the doc is very specific about when those exceptions should be thrown. The NPE is clearly not allowed to be thrown by `UnifiedSet`, because we allow null elements. The CCE is a bit peculiar, as any `Object` can fit in a `Set`, especially at runtime when generic types are erased. I suppose a badly implemented `equals` method could throw a CCE, but that would be because it's a badly implemented `equals` method, not because the logic in the set has determined that something doesn't fit.

Reply:
I've removed this change and the other two similar after verifying they all were indeed calling `this.containsAll` that delegates to `allSatisfyWith`. Thx!